src/swtpm/capabilities.c

```diff
@@ -122,19 +123,28 @@ int capabilities_print_json(bool cusetpm)
 #else
     const char *cmdarg_seccomp = "";
 #endif
+    const char *with_tpm1 = "";
+    const char *with_tpm2 = "";
     char *keysizecaps = NULL;
 
     ret = get_rsa_keysize_caps(&keysizecaps);
     if (ret < 0)
         goto cleanup;
 
+    if (TPMLIB_ChooseTPMVersion(TPMLIB_TPM_VERSION_1_2) == TPM_SUCCESS)
+        with_tpm1 = "\"tpm-1.2\", ";
```

Comment:
If this is now not showing tpm-1.2 then this would mean that no tpm-1.2 is available. Older versions of swtpm would not show that either, giving the impression that TPM 1.2 was not available. I think we need to reverse the logic here and show 'no-tpm-1.2' also because this is the new feature you added to libtpms.

Reply:
Why not version the capabilities? I think it it would be simpler. Introducing a negative capabilty is awkward and still something we cannot retrofit, so users will need to adjust depending on the version.

Reply:
Then let's try it with a version.

Reply:
You are right though, if we want to backport this, we should have some "no-tpm". Tbh, I think we shouldn't backport such major change.

Reply:
I didn't mean backporting but interpretation of this new verb now depends on the version identifier. If tpm-1.2 is missing prior to version 0.7 then that doesn't mean that TPM 1.2 isn't there.